backend-rust/src/graphql_api/account.rs

```diff
@@ -400,8 +400,15 @@ impl Account {
         Ok(slot_time)
     }
 
-    /// Number of transactions where this account is used as sender.
-    async fn transaction_count<'a>(&self, ctx: &Context<'a>) -> ApiResult<i64> {
+    /// Number of transactions the account has been involved in or
+    /// affected by.
+    async fn transaction_count<'a>(&self) -> ApiResult<i64> { Ok(self.num_txs) }
+
+    /// Number of transactions where this account is the sender of the
+    /// transaction. This is the `nonce` of the account. This value is
+    /// currently not used by the front-end and the `COUNT(*)` will need further
+    /// optimization if intended to be used.
+    async fn nonce<'a>(&self, ctx: &Context<'a>) -> ApiResult<i64> {
```

Comment:
Unclear to me why this is called a nonce but sure. :) 

Nonce is something that is supposed to be used for a one time purpose.

I do not see how that is the number of transactions.

Reply:
When an account sends a transaction and the transaction is finalized on the chain, the sequence number/nonce of that account is incremented. Hence, when we count the number of transactions where the queried account is the sender of the transaction, we get the same value as the `nonce` of the account.

Yes, it is called `nonce` because a transaction is signed and as such a transaction can only be finalized one time on-chain.

https://github.com/Concordium/concordium-node/blob/fc33d661178508b8eb7bcee6715169de7535ce59/concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs#L832